src/sentry/incidents/models.py

```diff
@@ -524,51 +529,29 @@ class Meta:
 
 
 @region_silo_only_model
-class AlertRuleTriggerAction(Model):
+class AlertRuleTriggerAction(AbstractNotificationAction):
     """
     This model represents an action that occurs when a trigger is fired. This is
     typically some sort of notification.
     """
 
+    # Aliases from NotificationAction introduction
+    Type = ActionServiceType
+    TargetType = ActionTargetType
+
     __include_in_export__ = True
 
     _type_registrations = {}
 
-    # Which sort of action to take
-    class Type(Enum):
-        EMAIL = 0
-        PAGERDUTY = 1
-        SLACK = 2
-        MSTEAMS = 3
-        SENTRY_APP = 4
-
     INTEGRATION_TYPES = frozenset((Type.PAGERDUTY.value, Type.SLACK.value, Type.MSTEAMS.value))
 
-    class TargetType(Enum):
-        # A direct reference, like an email address, Slack channel, or PagerDuty service
-        SPECIFIC = 0
-        # A specific user. This could be used to grab the user's email address.
-        USER = 1
-        # A specific team. This could be used to send an email to everyone associated
-        # with a team.
-        TEAM = 2
-        # A Sentry App instead of any of the above.
-        SENTRY_APP = 3
-
     TypeRegistration = namedtuple(
         "TypeRegistration",
         ["handler", "slug", "type", "supported_target_types", "integration_provider"],
     )
 
     alert_rule_trigger = FlexibleForeignKey("sentry.AlertRuleTrigger")
-    integration = FlexibleForeignKey("sentry.Integration", null=True)
-    sentry_app = FlexibleForeignKey("sentry.SentryApp", null=True)
-    type = models.SmallIntegerField()
-    target_type = models.SmallIntegerField()
```

Comment:
These two fields are changing to `BoundedIntegerFields` with the choices coming from the enums they were already using,

Reply:
This turned out to be a dangerous migration and is no longer releveant